src/app/pages/applications/forms/kubernetes-settings.component.ts

```diff
@@ -101,29 +101,53 @@ export class KubernetesSettingsComponent implements FormConfiguration {
     private dialogService: DialogService, private modalService: ModalService,
     private appService: ApplicationsService) { }
 
-  preInit(entityEdit: any) {
-    this.entityEdit = entityEdit;
-    const pool_control = _.find(this.fieldSets[0].config, { name: 'pool' });
-    this.appService.getPoolList().subscribe((pools) => {
-      pools.forEach((pool) => {
-        pool_control.options.push({ label: pool.name, value: pool.name });
+  async prerequisite(): Promise<boolean> {
+    const promise1 = new Promise((resolve, reject) => {
+      const pool_control = _.find(this.fieldSets[0].config, { name: 'pool' });
+      this.appService.getPoolList().toPromise().then((pools) => {
+        pools.forEach((pool) => {
+          pool_control.options.push({ label: pool.name, value: pool.name });
+        });
+        resolve(true);
+      },
+      (err) => {
+        resolve(false);
       });
     });
-    const node_ip_control = _.find(this.fieldSets[0].config, { name: 'node_ip' });
-    this.appService.getBindIPChoices().subscribe((ips) => {
-      for (const ip in ips) {
-        node_ip_control.options.push({ label: ip, value: ip });
-      }
+
+    const promise2 = new Promise((resolve, reject) => {
+      const node_ip_control = _.find(this.fieldSets[0].config, { name: 'node_ip' });
+      this.appService.getBindIPChoices().toPromise().then((ips) => {
+        for (const ip in ips) {
+          node_ip_control.options.push({ label: ip, value: ip });
+        }
+        resolve(true);
+      },
+      (err) => {
+        resolve(false);
+      });
     });
-    const v4_interface_control = _.find(this.fieldSets[1].config, { name: 'route_v4_interface' });
-    this.appService.getInterfaces().subscribe((interfaces: any[]) => {
-      interfaces.forEach((i) => {
-        v4_interface_control.options.push({ label: i.name, value: i.name });
+
+    const promise3 = new Promise((resolve, reject) => {
+      const v4_interface_control = _.find(this.fieldSets[1].config, { name: 'route_v4_interface' });
+      this.appService.getInterfaces().toPromise().then((interfaces: any[]) => {
+        interfaces.forEach((i) => {
+          v4_interface_control.options.push({ label: i.name, value: i.name });
+        });
+        resolve(true);
+      },
+      (err) => {
+        resolve(false);
       });
     });
+
+    return await Promise.all([promise1, promise2, promise3]).then(
+      (res) => true,
+    );
```

Comment:
I know that the `entity-form.interface.ts` uses promises for prerequisite, but I think we should be using observables when possible. They are more natural to Angular ecosystem and we will have less work, if we decide to start using ngrx.

Also, please correct naming of variables when doing so to make them descriptive and match camelCase:

Example:
```typescript
const setPoolControl = this.appService.getPoolList().pipe(
  tap((pools) => {
    const poolControl = _.find(this.fieldSets[0].config, { name: 'pool' });
    pools.forEach((pool) => {
      poolControl.options.push({ label: pool.name, value: pool.name });
    });
  }),
);

const setNodeIpControl = this.appService.getBindIPChoices().pipe(
  tap((ips) => {
    const nodeIpControl = _.find(this.fieldSets[0].config, { name: 'node_ip' });
    for (const ip of ips) {
      nodeIpControl.options.push({ label: ip, value: ip });
    }
  }),
);

...

return forkJoin([setPoolControl, setNodeIpControl, ...])
  .pipe(
    map(() => true),
    // catchError may not be needed if you want to fail when something hasn't been loaded.
    catchError((error) => {
      console.log(error);
      return of(false);
    }),
  )
  .toPromise();
```